acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -265,16 +289,20 @@ def rpc_blocking_request(broker, targets,
   end
 
   message = PCP::Message.new({
-    :message_type => 'http://puppetlabs.com/rpc_blocking_request',
+    :message_type => blocking ? 'http://puppetlabs.com/rpc_blocking_request' : 'http://puppetlabs.com/rpc_non_blocking_request',
     :targets => targets
   })
 
-  message.data = {
+  message_data = {
```

Comment:
For the sake of dealing with restarts, would it be useful to print the transaction_id? I think you should in case you can receive responses related to previous runs.

You're printing all responses (from the on_message handler), but you don't actually enforce that they're all related to the current transaction.